### PR TITLE
[bri-2] workflow setup pattern

### PR DESCRIPTION
**Pull request by bitwiseguy**
# Description

This PR introduces updates to bri-2 such that it can now create a new workflow with a unique zk-circuit, Shield smart contract, and zk Verifier smart contract by making a single request: `POST /workflows?type=signature`.

## Details

A key-manager + vault service developed by ConsenSys' Codefi team has been added. These services combine to provide RESTful endpoints to generate signing keys (Ethereum and eddsa keys), securely store those keys in a hashicorp vault, and perform digital signatures using those keys.

NATS is used for orchestrating the multi-step process to setup a new workflow. Since some of the steps include long-running processes, a pub/sub task manager is preferable to RESTful requests.

### CODE REVIEW

**Review by Meuko — APPROVED** [review]

File: examples/bri-2/commit-mgr/package.json [hunk 1]
```diff
@@ -1,63 +1,61 @@
...
+		"start": "node .",
+		"test": "jest --clearCache && jest --runInBand",
+		"up:besu": "docker-compose up -d alice-mongo alice-nats-server alice-besu",
+		"up:ganache": "docker-compose up -d alice-mongo alice-nats-server ganache",
+		"up:infura": "docker-compose up -d alice-mongo alice-nats-server",
+		"down": "docker-compose down"
+	},
+	"keywords": [],
+	"author": "",
+	"license": "ISC",
```
Comment (skarred14) [inline examples/bri-2/commit-mgr/package.json:21]: might want to update this license to CC0 or whichever is permissible

File: examples/bri-2/commit-mgr/src/blockchain/events.js [hunk 2]
```diff
@@ -1,72 +1,76 @@
...
+import { shieldContract } from './contracts';
 
 dotenv.config();
 
-export const newLeafEvent = ethers.utils.id("NewLeaf(uint256,bytes32,bytes32)");
+export const newLeafEvent = ethers.utils.id('NewLeaf(uint256,bytes32,bytes32)');
 
 export const subscribeMerkleEvents = (contractAddress) => {
-  logger.info(`Creating event listeners for contract: ${contractAddress}`);
+	logger.info(`Creating event listeners for contract: ${contractAddress}`);
```
Comment (skarred14) [inline examples/bri-2/commit-mgr/src/blockchain/events.js:16]: is this new indentation, the new linting format?

File: examples/bri-2/commit-mgr/src/index.ts [hunk 3]
```diff
@@ -1,60 +1,68 @@
...
+	await connectNATS();
+	await get_ws_provider(); // Establish websocket connection
+	await restartSubscriptions(); // Enable event listeners for active MerkleTrees
 
-  const app = express();
-  app.use(reqLogger('COMMIT-MGR')); // Log requests
-  app.use(reqErrorLogger('COMMIT-MGR')); // Log errors
-  app.use(bodyParser.json({ limit: "2mb" })); // Pre-parse body content
-  app.use(cors());
+	const app = express();
```
Comment (skarred14) [inline examples/bri-2/commit-mgr/src/index.ts:39]: same comment as above with reference to indentation

File: examples/bri-2/commit-mgr/src/db/models/Commit.js [hunk 4]
```diff
@@ -1,24 +1,33 @@
...
-    proof: [ Number ], // can use any three numbers for now
-    publicInputs: [ String ],
-    workflowStep: Number,
-    txHash: String,
-    creator: String, // i.e. Alice Eth address
-    signatures: Object, // mapping from participant id -> signature
-    status: String, // 'created', 'signed', 'layer-2', 'sent-mainnet', 'mainnet'
-  },
-  { versionKey: false }
+	{
```
Comment (skarred14) [inline examples/bri-2/commit-mgr/src/db/models/Commit.js:5]: same comment as above with reference to indentation

File: examples/bri-2/commit-mgr/src/blockchain/utils.js [hunk 5]
```diff
@@ -1,139 +1,142 @@
...
-      name = 'LOCAL';
-      break;
-    default:
-      name = 'LOCAL';
-      break;
-  }
-  return name;
-}
+	let name;
+	switch (parseInt(process.env.CHAIN_ID, 10)) {
```
Comment (skarred14) [inline examples/bri-2/commit-mgr/src/blockchain/utils.js:118]: same comment as above with reference to indentation

File: examples/bri-2/commit-mgr/src/blockchain/utils.js [hunk 6]
```diff
@@ -1,139 +1,142 @@
...
-    toBlock: "latest",
-    address: contractAddress,
-    topics: [newLeafEvent]
-  };
-  const res = await jsonrpc('eth_getLogs', [params]);
-  const logs = res.result;
+	const params = {
+		fromBlock: convertedBlockNum,
+		toBlock: 'latest',
+		address: contractAddress,
```
Comment (skarred14) [inline examples/bri-2/commit-mgr/src/blockchain/utils.js:81]: same comment as above with reference to indentation

File: examples/bri-2/commit-mgr/src/blockchain/utils.js [hunk 7]
```diff
@@ -1,139 +1,142 @@
...
-    } catch (err) {
-      logger.error(`[WEBSOCKET] Cannot establish connection: ${process.env.ETH_CLIENT_WS}`);
-    }
-  }
-  return ws_provider;
-}
+	if (!ws_provider) {
+		try {
+			ws_provider = new ethers.providers.WebSocketProvider(process.env.ETH_CLIENT_WS);
+			ws_provider._websocket.on('error', (error) => {
```
Comment (skarred14) [inline examples/bri-2/commit-mgr/src/blockchain/utils.js:19]: same comment as above with reference to indentation

File: examples/bri-2/commit-mgr/src/blockchain/events.js [hunk 8]
```diff
@@ -1,72 +1,76 @@
...
 
-  const singleLeafFilter = {
-    address: contractAddress,
-    topics: [newLeafEvent]
-  }
+	const singleLeafFilter = {
```
Comment (skarred14) [inline examples/bri-2/commit-mgr/src/blockchain/events.js:17]: same comment as above with reference to indentation

**Comment by bitwiseguy** [comment]
@skarred14 I added a top level .prettierrc file to enforce formatting across bri-2. Hopefully that addresses your comments.

**Review by Meuko — APPROVED** [review]

**Review by skarred14 — APPROVED** [review]

**Pull request merged**